packages/devextreme/js/__internal/grids/grid_core/header_filter/m_header_filter_core.ts

```diff
@@ -280,10 +280,10 @@ export class HeaderFilterView extends Modules.View {
     const $content = that._popupContainer.$content();
     const needShowSelectAllCheckbox = !options.isFilterBuilder && headerFilterOptions.allowSelectAll;
     const widgetOptions = {
-      searchEnabled: headerFilterOptions.search.enabled,
-      searchTimeout: headerFilterOptions.search.timeout,
-      searchEditorOptions: headerFilterOptions.search.editorOptions,
-      searchMode: headerFilterOptions.search.mode || '',
+      searchEnabled: headerFilterOptions?.search?.enabled,
```

Comment:
Minor.

Let's cast the ```searchEnabled``` to the ```boolean```

```suggestion
      searchEnabled: headerFilterOptions?.search?.enabled ?? false,
```

Reply:
Not connected to changes in this PR